src/translator/text-based-field.ts

```diff
@@ -95,4 +140,36 @@ export namespace TextBasedField {
 
         return textBasedFieldSchema;
     };
+
+    export const translateToElement = (element, key: string) => {
+        const { showIf, uiType, validation, ...rest } = element;
+        const requiredValidation = validation.filter(
+            (child: { hasOwnProperty: (arg0: string) => any }) =>
+                !Object.prototype.hasOwnProperty.call(child, "required")
+        );
```

Comment:
`requiredValidation` is not an accurate name because it is checking for non-required rules, moreover, you need to check for the existence for the required rule.

the current way of doing `required: validation?.[0]?.required` is not sufficient because the rule may not always be at the first entry.

so there are two things you want to achieve here:
1. check for the existence of required rule and ensuring it is true to pass to `required` property
2. find the remaining rules to pass to the `validation` array

instead of doing a filter, you can use `forEach` to construct the 2 variables.